### PR TITLE
Added "epoll" Linux executor - no PMR at all.

### PR DESCRIPTION
**Pull request by serges147**
- Added `example::Linux::EpollSingleThreadedExecutor` Linux executor - no PMR at all.
- Renamed `example::posix::PosixSingleThreadedExecutor` → `example::posix::PollSingleThreadedExecutor`.
- Eliminated CAVL tree from `PollSingleThreadedExecutor` - now double-linked-list is in use to automatically manage list of awaitable nodes (including their move); no virtual `SingleThreadedExecutor::onCallbackHandling` anymore.
- Added new `ITimeProvider` - provides current time via `TimePoint now()` method.

### CODE REVIEW

File: include/libcyphal/executor.hpp [hunk 1]
```diff
@@ -102,7 +102,7 @@ class IExecutor
         ///
         /// Size is chosen arbitrary, but it should be enough to store any callback implementation.
         ///
-        static constexpr std::size_t MaxSize = (sizeof(void*) * 12) + sizeof(Function);
+        static constexpr std::size_t MaxSize = (sizeof(void*) * 16) + sizeof(Function);
```
Comment (serges147) [inline include/libcyphal/executor.hpp:105]: Our `AwaitableNode`-s (which are inherited from `CallbackNode`) now hold couple extra fields (like `fd`, polling flags, etc), so unfortunately we need even bigger footprint to fit everything inside of `Callback::Any` unbounded variant - this is tradeoff for stack allocated callbacks.

File: include/libcyphal/executor.hpp [hunk 2]
```diff
@@ -127,9 +127,8 @@ class IExecutor
             /// according to the last setup.
             ///
             /// @param schedule Contains specifics of how exactly callback will be scheduled.
-            /// @return `true` if scheduling was successful, `false` otherwise (b/c callback has been reset).
             ///
-            virtual bool schedule(const Schedule::Variant& schedule) = 0;
+            virtual void schedule(const Schedule::Variant& schedule) = 0;
```
Comment (serges147) [inline include/libcyphal/executor.hpp:131]: Can't fail at all. Note that, in contrast, `Callback::Any::schedule` still has `bool` return value - the only reason is b/c of possible unbounded_variant without a value (`.has_value() == false`).

File: docs/examples/platform/posix/posix_single_threaded_executor.hpp [hunk 3]
```diff
@@ -36,42 +35,36 @@ namespace platform
...
+/// @brief Defines POSIX platform specific single-threaded executor based on `poll` mechanism.
+///
+class PollSingleThreadedExecutor final : public libcyphal::platform::SingleThreadedExecutor,
+                                         public IPosixExecutorExtension
 {
 public:
-    explicit PosixSingleThreadedExecutor(cetl::pmr::memory_resource& memory_resource)
-        : total_awaitables_{0}
-        , awaitable_nodes_allocator_{&memory_resource}
+    explicit PollSingleThreadedExecutor(cetl::pmr::memory_resource& memory_resource)
```
[the remark this answers is not in the document]
Comment (pavel-kirienko) [reply docs/examples/platform/posix/posix_single_threaded_executor.hpp:44]: image

File: include/libcyphal/executor.hpp (hunk 2 above)
Comment (pavel-kirienko) [reply include/libcyphal/executor.hpp:132]: <sub><sup>*failure is not an option!*</sub></sup>

File: docs/examples/platform/posix/posix_executor_extension.hpp [hunk 4]
```diff
@@ -21,40 +21,43 @@ namespace posix
...
             int fd;
         };
-        struct HandleWritable
+        struct Writable
         {
             int fd;
         };
 
-        using Variant = cetl::variant<HandleReadable, HandleWritable>;
+        using Variant = cetl::variant<Readable, Writable>;
```
Comment (pavel-kirienko) [inline docs/examples/platform/posix/posix_executor_extension.hpp:45]: Much nicer

**Review by pavel-kirienko — APPROVED** [review]

File: docs/examples/platform/posix/posix_executor_extension.hpp (hunk 4 above)
Comment (serges147) [reply docs/examples/platform/posix/posix_executor_extension.hpp:45]: will fix it in a near future pr.

**Review by lydia-at-amazon — APPROVED** [review]

**Pull request merged**